## What you are we answering:

<br>How does Kayali's price per ml compare to competitors in each market?</br>
<br>Where is Kayali priced above or below the market midpoint?</br>
<br>Is there a pricing gap between Kayali's own site and retailers?</br> 

# Installing the libraries

pip install openpyxl

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import openpyxl as op

In [2]:
pricing_df=pd.read_excel("Pricing Data.xlsx", sheet_name="Pricing_Data")

In [3]:
market_df=pd.read_excel("Market Data.xlsx",sheet_name="Market_Data")

In [4]:
sales_df=pd.read_excel("Sales Data.xlsx",sheet_name="Sales_Data")

In [5]:
pricing_df.head()

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD,Price_own_GBP,Price_Retailer_GBP,Price_Gap_GBP,Retailer_Name_GBP,Price_own_AED,Price_Retailer_UAE,Price_Gap_UAE,Retailer_Name_UAE,On_Promotion,Date_Collected,Notes
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,105.0,105.0,0.0,Sephora,80.0,80.0,0.0,Sephora,420,420.0,0.0,Sephora,No,2026-04-20,NaN
1,Kayali,Eden Juicy Apple 01,Eden collection,Kayali,50ml,105.0,105.0,0.0,Sephora,80.0,80.0,0.0,Sephora,420,420.0,0.0,Sephora,No,2026-04-20,NaN
2,Kayali,Oudgasm Tobacco Oud 04,Oudgasm collection,Kayali,50ml,140.0,119.0,21.0,Sephora,119.0,119.0,0.0,Sephora,551,551.0,0.0,Sephora,No,2026-04-20,NaN
3,Kayali,Yum Pistachio Gelato 33,Yum collection,Kayali,50ml,105.0,105.0,0.0,Sephora,80.0,80.0,0.0,Sephora,420,420.0,0.0,Sephora,No,2026-04-20,NaN
4,Kayali,Freedom Musk Santal 34,Freedom collection,Kayali,50ml,105.0,105.0,0.0,Sephora,80.0,80.0,0.0,Sephora,420,420.0,0.0,Sephora,No,2026-04-20,NaN


In [6]:
pricing_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18 entries, 0 to 17
Data columns (total 20 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Brand               18 non-null     object        
 1   Fragrance_Name      18 non-null     object        
 2   Collection          15 non-null     object        
 3   Type                18 non-null     object        
 4   Size_ml             18 non-null     object        
 5   Price_own_USD       17 non-null     float64       
 6   Price_retailer_USD  15 non-null     float64       
 7   Price_Gap_USD       15 non-null     float64       
 8   Retailer_Name_USD   18 non-null     object        
 9   Price_own_GBP       14 non-null     float64       
 10  Price_Retailer_GBP  16 non-null     float64       
 11  Price_Gap_GBP       13 non-null     float64       
 12  Retailer_Name_GBP   18 non-null     object        
 13  Price_own_AED       18 non-null     int64         
 

In [7]:
pricing_df.describe()

,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Price_own_GBP,Price_Retailer_GBP,Price_Gap_GBP,Price_own_AED,Price_Retailer_UAE,Price_Gap_UAE,Date_Collected
count,17.000000,15.000000,15.000000,14.000000,16.000000,13.000000,18.000000,17.000000,17.000000,18
mean,157.176471,137.000000,11.466667,128.285714,114.187500,2.846154,584.333333,584.411765,9.588235,2026-04-20 12:00:00
min,44.000000,22.000000,-9.000000,80.000000,20.000000,0.000000,119.000000,105.000000,0.000000,2026-04-20 00:00:00
25%,105.000000,105.000000,0.000000,80.000000,80.000000,0.000000,420.000000,420.000000,0.000000,2026-04-20 00:00:00
50%,140.000000,119.000000,0.000000,120.500000,111.000000,0.000000,530.000000,535.000000,0.000000,2026-04-20 12:00:00
75%,170.000000,170.000000,21.500000,148.000000,132.000000,0.000000,745.000000,745.000000,0.000000,2026-04-21 00:00:00
max,340.000000,349.000000,78.000000,238.000000,238.000000,27.000000,1322.000000,1322.000000,105.000000,2026-04-21 00:00:00
std,87.275595,77.161241,23.108028,53.667679,60.972364,7.492732,333.951212,352.670182,26.206056,NaN


In [8]:
pricing_df

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD,Price_own_GBP,Price_Retailer_GBP,Price_Gap_GBP,Retailer_Name_GBP,Price_own_AED,Price_Retailer_UAE,Price_Gap_UAE,Retailer_Name_UAE,On_Promotion,Date_Collected,Notes
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,105.0,105.0,0.0,Sephora,80.0,80.0,0.0,Sephora,420,420.0,0.0,Sephora,No,2026-04-20,NaN
1,Kayali,Eden Juicy Apple 01,Eden collection,Kayali,50ml,105.0,105.0,0.0,Sephora,80.0,80.0,0.0,Sephora,420,420.0,0.0,Sephora,No,2026-04-20,NaN
2,Kayali,Oudgasm Tobacco Oud 04,Oudgasm collection,Kayali,50ml,140.0,119.0,21.0,Sephora,119.0,119.0,0.0,Sephora,551,551.0,0.0,Sephora,No,2026-04-20,NaN
3,Kayali,Yum Pistachio Gelato 33,Yum collection,Kayali,50ml,105.0,105.0,0.0,Sephora,80.0,80.0,0.0,Sephora,420,420.0,0.0,Sephora,No,2026-04-20,NaN
4,Kayali,Freedom Musk Santal 34,Freedom collection,Kayali,50ml,105.0,105.0,0.0,Sephora,80.0,80.0,0.0,Sephora,420,420.0,0.0,Sephora,No,2026-04-20,NaN
5,Kayali,Musk 12,Vanilla collection,Kayali,50ml,105.0,NaN,NaN,Sephora,80.0,NaN,NaN,Sephora,420,NaN,NaN,Sephora,No,2026-04-20,This one's website exclusive
6,Jo Malone,English Pear & Freesia,Cologne,Competitor,100 ml,170.0,170.0,0.0,Sephora,122.0,122.0,0.0,John Lewis,595,595.0,0.0,Sephora,No,2026-04-20,NaN
7,Jo Malone,Myrrh & Tonka,Cologne Intense,Competitor,50 ml,160.0,160.0,0.0,Sephora,116.0,116.0,0.0,John Lewis,535,535.0,0.0,Sephora,No,2026-04-20,NaN
8,Maison Margiela Replica,By the Fireplace,Replica Collection,Competitor,100 ml,170.0,170.0,0.0,Sephora,126.0,126.0,0.0,Sephora,745,745.0,0.0,Sephora,No,2026-04-20,NaN
9,Maison Margiela Replica,Jazz Club,Replica Collection,Competitor,100 ml,170.0,170.0,0.0,Sephora,127.0,100.0,27.0,Sephora,745,745.0,0.0,Sephora,Yes for GBP,2026-04-21,NaN


In [ ]:
# Split the pricing data into 3 DF's --> USA, UK, UAE

In [9]:
usa_df=pricing_df[['Brand','Fragrance_Name','Collection','Type','Size_ml','Price_own_USD','Price_retailer_USD','Price_Gap_USD','Retailer_Name_USD']]

In [10]:
usa_df

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,105.0,105.0,0.0,Sephora
1,Kayali,Eden Juicy Apple 01,Eden collection,Kayali,50ml,105.0,105.0,0.0,Sephora
2,Kayali,Oudgasm Tobacco Oud 04,Oudgasm collection,Kayali,50ml,140.0,119.0,21.0,Sephora
3,Kayali,Yum Pistachio Gelato 33,Yum collection,Kayali,50ml,105.0,105.0,0.0,Sephora
4,Kayali,Freedom Musk Santal 34,Freedom collection,Kayali,50ml,105.0,105.0,0.0,Sephora
5,Kayali,Musk 12,Vanilla collection,Kayali,50ml,105.0,NaN,NaN,Sephora
6,Jo Malone,English Pear & Freesia,Cologne,Competitor,100 ml,170.0,170.0,0.0,Sephora
7,Jo Malone,Myrrh & Tonka,Cologne Intense,Competitor,50 ml,160.0,160.0,0.0,Sephora
8,Maison Margiela Replica,By the Fireplace,Replica Collection,Competitor,100 ml,170.0,170.0,0.0,Sephora
9,Maison Margiela Replica,Jazz Club,Replica Collection,Competitor,100 ml,170.0,170.0,0.0,Sephora


In [11]:
usa_df['Best_price_per_ml']=usa_df['Price_own_USD'].fillna(usa_df['Price_retailer_USD'])

<ipython-input-11-1c234c4fa30f>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  usa_df['Best_price_per_ml']=usa_df['Price_own_USD'].fillna(usa_df['Price_retailer_USD'])


In [12]:
usa_df

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD,Best_price_per_ml
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,105.0,105.0,0.0,Sephora,105.0
1,Kayali,Eden Juicy Apple 01,Eden collection,Kayali,50ml,105.0,105.0,0.0,Sephora,105.0
2,Kayali,Oudgasm Tobacco Oud 04,Oudgasm collection,Kayali,50ml,140.0,119.0,21.0,Sephora,140.0
3,Kayali,Yum Pistachio Gelato 33,Yum collection,Kayali,50ml,105.0,105.0,0.0,Sephora,105.0
4,Kayali,Freedom Musk Santal 34,Freedom collection,Kayali,50ml,105.0,105.0,0.0,Sephora,105.0
5,Kayali,Musk 12,Vanilla collection,Kayali,50ml,105.0,NaN,NaN,Sephora,105.0
6,Jo Malone,English Pear & Freesia,Cologne,Competitor,100 ml,170.0,170.0,0.0,Sephora,170.0
7,Jo Malone,Myrrh & Tonka,Cologne Intense,Competitor,50 ml,160.0,160.0,0.0,Sephora,160.0
8,Maison Margiela Replica,By the Fireplace,Replica Collection,Competitor,100 ml,170.0,170.0,0.0,Sephora,170.0
9,Maison Margiela Replica,Jazz Club,Replica Collection,Competitor,100 ml,170.0,170.0,0.0,Sephora,170.0


In [13]:
usa_df=usa_df[usa_df['Best_price_per_ml'].notna()]

In [14]:
usa_df

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD,Best_price_per_ml
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,105.0,105.0,0.0,Sephora,105.0
1,Kayali,Eden Juicy Apple 01,Eden collection,Kayali,50ml,105.0,105.0,0.0,Sephora,105.0
2,Kayali,Oudgasm Tobacco Oud 04,Oudgasm collection,Kayali,50ml,140.0,119.0,21.0,Sephora,140.0
3,Kayali,Yum Pistachio Gelato 33,Yum collection,Kayali,50ml,105.0,105.0,0.0,Sephora,105.0
4,Kayali,Freedom Musk Santal 34,Freedom collection,Kayali,50ml,105.0,105.0,0.0,Sephora,105.0
5,Kayali,Musk 12,Vanilla collection,Kayali,50ml,105.0,NaN,NaN,Sephora,105.0
6,Jo Malone,English Pear & Freesia,Cologne,Competitor,100 ml,170.0,170.0,0.0,Sephora,170.0
7,Jo Malone,Myrrh & Tonka,Cologne Intense,Competitor,50 ml,160.0,160.0,0.0,Sephora,160.0
8,Maison Margiela Replica,By the Fireplace,Replica Collection,Competitor,100 ml,170.0,170.0,0.0,Sephora,170.0
9,Maison Margiela Replica,Jazz Club,Replica Collection,Competitor,100 ml,170.0,170.0,0.0,Sephora,170.0


In [15]:
usa_df.shape

(17, 10)

In [16]:
uk_df=pricing_df[['Brand','Fragrance_Name','Collection','Type','Size_ml','Price_own_GBP','Price_Retailer_GBP','Price_Gap_GBP','Retailer_Name_GBP']]

In [17]:
uk_df.shape

(18, 9)

In [18]:
uk_df['Best_price_per_ml']=uk_df['Price_own_GBP'].fillna(uk_df['Price_Retailer_GBP'])

<ipython-input-18-c149800f09e3>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  uk_df['Best_price_per_ml']=uk_df['Price_own_GBP'].fillna(uk_df['Price_Retailer_GBP'])


In [19]:
uk_df=uk_df[uk_df['Best_price_per_ml'].notna()]

In [20]:
uk_df.shape

(17, 10)

In [21]:
uk_df

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_GBP,Price_Retailer_GBP,Price_Gap_GBP,Retailer_Name_GBP,Best_price_per_ml
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,80.0,80.0,0.0,Sephora,80.0
1,Kayali,Eden Juicy Apple 01,Eden collection,Kayali,50ml,80.0,80.0,0.0,Sephora,80.0
2,Kayali,Oudgasm Tobacco Oud 04,Oudgasm collection,Kayali,50ml,119.0,119.0,0.0,Sephora,119.0
3,Kayali,Yum Pistachio Gelato 33,Yum collection,Kayali,50ml,80.0,80.0,0.0,Sephora,80.0
4,Kayali,Freedom Musk Santal 34,Freedom collection,Kayali,50ml,80.0,80.0,0.0,Sephora,80.0
5,Kayali,Musk 12,Vanilla collection,Kayali,50ml,80.0,NaN,NaN,Sephora,80.0
6,Jo Malone,English Pear & Freesia,Cologne,Competitor,100 ml,122.0,122.0,0.0,John Lewis,122.0
7,Jo Malone,Myrrh & Tonka,Cologne Intense,Competitor,50 ml,116.0,116.0,0.0,John Lewis,116.0
8,Maison Margiela Replica,By the Fireplace,Replica Collection,Competitor,100 ml,126.0,126.0,0.0,Sephora,126.0
9,Maison Margiela Replica,Jazz Club,Replica Collection,Competitor,100 ml,127.0,100.0,27.0,Sephora,127.0


In [22]:
uae_df=pricing_df[['Brand','Fragrance_Name','Collection','Type','Size_ml','Price_own_AED','Price_Retailer_UAE','Price_Gap_UAE','Retailer_Name_UAE']]

In [23]:
uae_df

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_AED,Price_Retailer_UAE,Price_Gap_UAE,Retailer_Name_UAE
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,420,420.0,0.0,Sephora
1,Kayali,Eden Juicy Apple 01,Eden collection,Kayali,50ml,420,420.0,0.0,Sephora
2,Kayali,Oudgasm Tobacco Oud 04,Oudgasm collection,Kayali,50ml,551,551.0,0.0,Sephora
3,Kayali,Yum Pistachio Gelato 33,Yum collection,Kayali,50ml,420,420.0,0.0,Sephora
4,Kayali,Freedom Musk Santal 34,Freedom collection,Kayali,50ml,420,420.0,0.0,Sephora
5,Kayali,Musk 12,Vanilla collection,Kayali,50ml,420,NaN,NaN,Sephora
6,Jo Malone,English Pear & Freesia,Cologne,Competitor,100 ml,595,595.0,0.0,Sephora
7,Jo Malone,Myrrh & Tonka,Cologne Intense,Competitor,50 ml,535,535.0,0.0,Sephora
8,Maison Margiela Replica,By the Fireplace,Replica Collection,Competitor,100 ml,745,745.0,0.0,Sephora
9,Maison Margiela Replica,Jazz Club,Replica Collection,Competitor,100 ml,745,745.0,0.0,Sephora


In [24]:
uae_df.shape

(18, 9)

In [25]:
uae_df['Best_price_per_ml']=uae_df['Price_own_AED'].fillna(uae_df['Price_Retailer_UAE'])


<ipython-input-25-5f14f0e830ae>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  uae_df['Best_price_per_ml']=uae_df['Price_own_AED'].fillna(uae_df['Price_Retailer_UAE'])


In [26]:
uae_df=uae_df[uae_df['Best_price_per_ml'].notna()]

In [27]:
uae_df

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_AED,Price_Retailer_UAE,Price_Gap_UAE,Retailer_Name_UAE,Best_price_per_ml
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,420,420.0,0.0,Sephora,420
1,Kayali,Eden Juicy Apple 01,Eden collection,Kayali,50ml,420,420.0,0.0,Sephora,420
2,Kayali,Oudgasm Tobacco Oud 04,Oudgasm collection,Kayali,50ml,551,551.0,0.0,Sephora,551
3,Kayali,Yum Pistachio Gelato 33,Yum collection,Kayali,50ml,420,420.0,0.0,Sephora,420
4,Kayali,Freedom Musk Santal 34,Freedom collection,Kayali,50ml,420,420.0,0.0,Sephora,420
5,Kayali,Musk 12,Vanilla collection,Kayali,50ml,420,NaN,NaN,Sephora,420
6,Jo Malone,English Pear & Freesia,Cologne,Competitor,100 ml,595,595.0,0.0,Sephora,595
7,Jo Malone,Myrrh & Tonka,Cologne Intense,Competitor,50 ml,535,535.0,0.0,Sephora,535
8,Maison Margiela Replica,By the Fireplace,Replica Collection,Competitor,100 ml,745,745.0,0.0,Sephora,745
9,Maison Margiela Replica,Jazz Club,Replica Collection,Competitor,100 ml,745,745.0,0.0,Sephora,745


In [28]:
uae_df.shape

(18, 10)

In [29]:
usa_df.head(1)

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD,Best_price_per_ml
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,105.0,105.0,0.0,Sephora,105.0


In [30]:
usa_df.rename(columns={'Best_price_per_ml':'Best_price'},inplace=True)

<ipython-input-30-91a386661742>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  usa_df.rename(columns={'Best_price_per_ml':'Best_price'},inplace=True)


In [31]:
usa_df.head(1)

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD,Best_price
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,105.0,105.0,0.0,Sephora,105.0


In [32]:
uk_df.head(1)

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_GBP,Price_Retailer_GBP,Price_Gap_GBP,Retailer_Name_GBP,Best_price_per_ml
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,80.0,80.0,0.0,Sephora,80.0


In [33]:
uk_df.rename(columns={'Best_price_per_ml':'Best_price'},inplace=True)

In [34]:
uk_df.head(1)

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_GBP,Price_Retailer_GBP,Price_Gap_GBP,Retailer_Name_GBP,Best_price
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,80.0,80.0,0.0,Sephora,80.0


In [35]:
uae_df.head(1)

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_AED,Price_Retailer_UAE,Price_Gap_UAE,Retailer_Name_UAE,Best_price_per_ml
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,420,420.0,0.0,Sephora,420


In [36]:
uae_df.rename(columns={'Best_price_per_ml':'Best_price'},inplace=True)

In [37]:
uae_df.head(1)

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_AED,Price_Retailer_UAE,Price_Gap_UAE,Retailer_Name_UAE,Best_price
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50ml,420,420.0,0.0,Sephora,420


In [38]:
usa_df['Size_ml']=pd.to_numeric(usa_df['Size_ml'].str.replace("ml","",regex=False).str.strip())

<ipython-input-38-91b3a93fa664>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  usa_df['Size_ml']=pd.to_numeric(usa_df['Size_ml'].str.replace("ml","",regex=False).str.strip())


In [ ]:
#Price per ml is calculated for all 3 markets

In [39]:
usa_df['Best_price_per_ml_USD']=usa_df['Best_price']/usa_df['Size_ml']

<ipython-input-39-bbf6b5ca4f57>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  usa_df['Best_price_per_ml_USD']=usa_df['Best_price']/usa_df['Size_ml']


In [40]:
usa_df.head(1)

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD,Best_price,Best_price_per_ml_USD
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50,105.0,105.0,0.0,Sephora,105.0,2.1


In [41]:
uk_df['Size_ml']=pd.to_numeric(uk_df['Size_ml'].str.replace("ml","",regex=False).str.strip())

In [42]:
uk_df['Best_price_per_ml_GBP']=uk_df['Best_price']/uk_df['Size_ml']

In [43]:
uae_df['Size_ml']=pd.to_numeric(uae_df['Size_ml'].str.replace("ml","",regex=False).str.strip())

In [44]:
uae_df['Best_price_per_ml_AED']=uae_df['Best_price']/uae_df['Size_ml']

In [45]:
pricing_df.dtypes

Brand                         object
Fragrance_Name                object
Collection                    object
Type                          object
Size_ml                       object
Price_own_USD                float64
Price_retailer_USD           float64
Price_Gap_USD                float64
Retailer_Name_USD             object
Price_own_GBP                float64
Price_Retailer_GBP           float64
Price_Gap_GBP                float64
Retailer_Name_GBP             object
Price_own_AED                  int64
Price_Retailer_UAE           float64
Price_Gap_UAE                float64
Retailer_Name_UAE             object
On_Promotion                  object
Date_Collected        datetime64[ns]
Notes                         object
dtype: object

In [46]:
print("USA rows:",usa_df.shape[0])
print("UK rows:",uk_df.shape[0])
print("UAE rows:",uae_df.shape[0])

USA rows: 17
UK rows: 17
UAE rows: 18


<br> What is the one row difference between these countries? Which fragrance & brand has no price in their own site & their retailer? : Electric Oud from Ajmal is not available in US & UK <br>

In [47]:
usa_df.head(1)

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD,Best_price,Best_price_per_ml_USD
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50,105.0,105.0,0.0,Sephora,105.0,2.1


In [48]:
kayali_usa=usa_df[usa_df['Type']=='Kayali']

In [49]:
usa_df['Type'].unique()

array(['Kayali', 'Competitor'], dtype=object)

In [50]:
competitor_usa=usa_df[usa_df['Type']=='Competitor']

In [51]:
#UK
kayali_uk=uk_df[uk_df['Type']=='Kayali']
competitor_uk=uk_df[uk_df['Type']=='Competitor']

In [52]:
#uae
kayali_uae=uae_df[uae_df['Type']=='Kayali']
competitor_uae=uae_df[uae_df['Type']=='Competitor']

In [53]:
print("Kayali US SKUs:", kayali_usa["Fragrance_Name"].nunique())
print("Competitor US products:", competitor_usa["Fragrance_Name"].nunique())

Kayali US SKUs: 6
Competitor US products: 11


In [54]:
print("Kayali UK SKUs:", kayali_uk["Fragrance_Name"].nunique())
print("Competitor UK products:", competitor_uk["Fragrance_Name"].nunique())

Kayali US SKUs: 6
Competitor US products: 11


In [55]:
print("Kayali UAE SKUs:", kayali_uae["Fragrance_Name"].nunique())
print("Competitor UAE products:", competitor_uae["Fragrance_Name"].nunique())

Kayali US SKUs: 6
Competitor US products: 12


In [56]:
kayali_usa

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD,Best_price,Best_price_per_ml_USD
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50,105.0,105.0,0.0,Sephora,105.0,2.1
1,Kayali,Eden Juicy Apple 01,Eden collection,Kayali,50,105.0,105.0,0.0,Sephora,105.0,2.1
2,Kayali,Oudgasm Tobacco Oud 04,Oudgasm collection,Kayali,50,140.0,119.0,21.0,Sephora,140.0,2.8
3,Kayali,Yum Pistachio Gelato 33,Yum collection,Kayali,50,105.0,105.0,0.0,Sephora,105.0,2.1
4,Kayali,Freedom Musk Santal 34,Freedom collection,Kayali,50,105.0,105.0,0.0,Sephora,105.0,2.1
5,Kayali,Musk 12,Vanilla collection,Kayali,50,105.0,NaN,NaN,Sephora,105.0,2.1


Calculate average competitor price per ml for each market:

In [58]:
avg_competitor_usd=competitor_usa['Best_price_per_ml_USD'].mean()

In [59]:
avg_competitor_gbp=competitor_uk['Best_price_per_ml_GBP'].mean()

In [60]:
avg_competitor_aed=competitor_uae['Best_price_per_ml_AED'].mean()

In [61]:
print(f"Average Competitor Price/ml USD: ${avg_competitor_usd:.2f}")
print(f"Average Competitor Price/ml GBP: £{avg_competitor_gbp:.2f}")
print(f"Average Competitor Price/ml AED: AED {avg_competitor_aed:.2f}")

Average Competitor Price/ml USD: $2.40
Average Competitor Price/ml GBP: £1.68
Average Competitor Price/ml AED: AED 8.32


Calculate Price Index for each Kayali SKU per market:

What is Price Index?
Price Index is simply a number that tells you how expensive Kayali is compared to the average competitor in that market.
It is always expressed relative to 100 where:

100 = Kayali is priced exactly at the competitor average
Above 100 = Kayali is more expensive than the average competitor
Below 100 = Kayali is cheaper than the average competitor


Simple Real World Example
Say in the US market the average competitor price per ml is $2.00
Kayali Vanilla 28 is priced at $2.10 per ml
Price Index = (2.10 ÷ 2.00) × 100 = 105
Price Index of 105 means Kayali Vanilla 28 is 5% more expensive than the average competitor in the US.

Now say Kayali Musk 12 is priced at $1.80 per ml
Price Index = (1.80 ÷ 2.00) × 100 = 90
Price Index of 90 means Kayali Musk 12 is 10% cheaper than the average competitor in the US.

In [62]:
kayali_usa['Price_Index']=(kayali_usa['Best_price_per_ml_USD']/avg_competitor_usd)*100

<ipython-input-62-56f29f555b4b>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  kayali_usa['Price_Index']=(kayali_usa['Best_price_per_ml_USD']/avg_competitor_usd)*100


In [63]:
kayali_uk['Price_Index']=(kayali_uk['Best_price_per_ml_GBP']/avg_competitor_gbp)*100

<ipython-input-63-db9096ab969c>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  kayali_uk['Price_Index']=(kayali_uk['Best_price_per_ml_GBP']/avg_competitor_gbp)*100


In [64]:
kayali_uae['Price_Index']=(kayali_uae['Best_price_per_ml_AED']/avg_competitor_aed)*100

<ipython-input-64-08baa848e6cc>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  kayali_uae['Price_Index']=(kayali_uae['Best_price_per_ml_AED']/avg_competitor_aed)*100


In [65]:
kayali_usa

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD,Best_price,Best_price_per_ml_USD,Price_Index
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50,105.0,105.0,0.0,Sephora,105.0,2.1,87.599545
1,Kayali,Eden Juicy Apple 01,Eden collection,Kayali,50,105.0,105.0,0.0,Sephora,105.0,2.1,87.599545
2,Kayali,Oudgasm Tobacco Oud 04,Oudgasm collection,Kayali,50,140.0,119.0,21.0,Sephora,140.0,2.8,116.799393
3,Kayali,Yum Pistachio Gelato 33,Yum collection,Kayali,50,105.0,105.0,0.0,Sephora,105.0,2.1,87.599545
4,Kayali,Freedom Musk Santal 34,Freedom collection,Kayali,50,105.0,105.0,0.0,Sephora,105.0,2.1,87.599545
5,Kayali,Musk 12,Vanilla collection,Kayali,50,105.0,NaN,NaN,Sephora,105.0,2.1,87.599545


In [66]:
kayali_uk

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_GBP,Price_Retailer_GBP,Price_Gap_GBP,Retailer_Name_GBP,Best_price,Best_price_per_ml_GBP,Price_Index
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50,80.0,80.0,0.0,Sephora,80.0,1.60,95.083739
1,Kayali,Eden Juicy Apple 01,Eden collection,Kayali,50,80.0,80.0,0.0,Sephora,80.0,1.60,95.083739
2,Kayali,Oudgasm Tobacco Oud 04,Oudgasm collection,Kayali,50,119.0,119.0,0.0,Sephora,119.0,2.38,141.437061
3,Kayali,Yum Pistachio Gelato 33,Yum collection,Kayali,50,80.0,80.0,0.0,Sephora,80.0,1.60,95.083739
4,Kayali,Freedom Musk Santal 34,Freedom collection,Kayali,50,80.0,80.0,0.0,Sephora,80.0,1.60,95.083739
5,Kayali,Musk 12,Vanilla collection,Kayali,50,80.0,NaN,NaN,Sephora,80.0,1.60,95.083739


Step 7: Combine all 3 Kayali market dataframes into one for charting:

In [67]:
kayali_usa['Market']='USA'
kayali_uk['Market']='UK'
kayali_uae['Market']='UAE'

<ipython-input-67-8f05ccfcab39>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  kayali_usa['Market']='USA'
<ipython-input-67-8f05ccfcab39>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  kayali_uk['Market']='UK'
<ipython-input-67-8f05ccfcab39>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#return

In [68]:
kayali_usa.head(1)

,Brand,Fragrance_Name,Collection,Type,Size_ml,Price_own_USD,Price_retailer_USD,Price_Gap_USD,Retailer_Name_USD,Best_price,Best_price_per_ml_USD,Price_Index,Market
0,Kayali,Vanilla 28,Vanilla collection,Kayali,50,105.0,105.0,0.0,Sephora,105.0,2.1,87.599545,USA


In [69]:
kayali_combined=pd.concat([
    kayali_usa[['Fragrance_Name','Market','Price_Index']],
    kayali_uk[['Fragrance_Name','Market','Price_Index']],
    kayali_uae[['Fragrance_Name','Market','Price_Index']]
])

In [70]:
kayali_combined

,Fragrance_Name,Market,Price_Index
0,Vanilla 28,USA,87.599545
1,Eden Juicy Apple 01,USA,87.599545
2,Oudgasm Tobacco Oud 04,USA,116.799393
3,Yum Pistachio Gelato 33,USA,87.599545
4,Freedom Musk Santal 34,USA,87.599545
5,Musk 12,USA,87.599545
0,Vanilla 28,UK,95.083739
1,Eden Juicy Apple 01,UK,95.083739
2,Oudgasm Tobacco Oud 04,UK,141.437061
3,Yum Pistachio Gelato 33,UK,95.083739


In [71]:
kayali_combined.to_excel('Kayali_Price_Index.xlsx')

<ipython-input-71-f7ac60eade2b>:1: UserWarning: Pandas requires version '1.4.3' or newer of 'xlsxwriter' (version '1.2.9' currently installed).
  kayali_combined.to_excel('Kayali_Price_Index.xlsx')


# #Retailer Price Gap Analysis

This section only uses rows where BOTH own site price AND retailer price exist. So we filter from the market dataframes not the Kayali-only ones — because wewant to see gaps for all brands not just Kayali.

In [74]:
gap_usa=usa_df[usa_df['Price_own_USD'].notna() & usa_df['Price_retailer_USD'].notna()].copy()

In [75]:
gap_uk=uk_df[uk_df['Price_own_GBP'].notna() & uk_df['Price_Retailer_GBP'].notna()].copy()

In [76]:
gap_uae=uae_df[uae_df['Price_own_AED'].notna() & uae_df['Price_Retailer_UAE'].notna()].copy()

In [77]:
gap_usa.shape

(15, 11)

In [78]:
gap_uk.shape

(13, 11)

In [79]:
gap_uae.shape

(17, 11)

In [80]:
def gap_direction(x):
    if x>0:
        return "Own site is Higher"
    elif x<0:
        return "Retailer Higher"
    else:
        return "Price Consistent"

In [81]:
gap_direction(0)

'Price Consistent'

In [82]:
gap_uae['Gap_Direction']=gap_uae['Price_Gap_UAE'].apply(gap_direction)

In [83]:
gap_uk['Gap_Direction']=gap_uk['Price_Gap_GBP'].apply(gap_direction)

In [84]:
gap_usa['Gap_Direction']=gap_usa['Price_Gap_USD'].apply(gap_direction)

Step 11: Build summary table combining all 3 markets:

In [85]:
gap_uae['Market']='UAE'
gap_uk['Market']='UK'
gap_usa['Market']='USA'

In [87]:
gap_usa_clean=gap_usa[['Brand','Fragrance_Name','Market','Price_own_USD','Price_retailer_USD','Price_Gap_USD','Gap_Direction']]

In [88]:
gap_uk_clean=gap_uk[['Brand','Fragrance_Name','Market','Price_own_GBP','Price_Retailer_GBP','Price_Gap_GBP','Gap_Direction']]

In [89]:
gap_uae_clean=gap_uae[['Brand','Fragrance_Name','Market','Price_own_AED','Price_Retailer_UAE','Price_Gap_UAE','Gap_Direction']]

In [90]:
gap_usa_clean.rename(columns={"Price_own_USD": "Own_Site_Price",
    "Price_retailer_USD": "Retailer_Price",
    "Price_Gap_USD": "Gap"},inplace=True)

<ipython-input-90-56bdf1986476>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  gap_usa_clean.rename(columns={"Price_own_USD": "Own_Site_Price",


In [91]:
gap_uk_clean.rename(columns={"Price_own_GBP": "Own_Site_Price",
    "Price_Retailer_GBP": "Retailer_Price",
    "Price_Gap_GBP": "Gap"},inplace=True)

<ipython-input-91-3af220c2bdda>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  gap_uk_clean.rename(columns={"Price_own_GBP": "Own_Site_Price",


In [92]:
gap_uae_clean.rename(columns={"Price_own_AED": "Own_Site_Price",
    "Price_Retailer_UAE": "Retailer_Price",
    "Price_Gap_UAE": "Gap"},inplace=True)

<ipython-input-92-1eba2e565f61>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  gap_uae_clean.rename(columns={"Price_own_AED": "Own_Site_Price",


In [93]:
gap_summary = pd.concat([gap_usa_clean, gap_uk_clean, gap_uae_clean], ignore_index=True)

"Price gap analysis conducted using full bottle prices — gap direction and magnitude are consistent indicators of channel pricing consistency regardless of bottle size."

In [94]:
gap_summary

,Brand,Fragrance_Name,Market,Own_Site_Price,Retailer_Price,Gap,Gap_Direction
0,Kayali,Vanilla 28,USA,105.0,105.0,0.0,Price Consistent
1,Kayali,Eden Juicy Apple 01,USA,105.0,105.0,0.0,Price Consistent
2,Kayali,Oudgasm Tobacco Oud 04,USA,140.0,119.0,21.0,Own site is Higher
3,Kayali,Yum Pistachio Gelato 33,USA,105.0,105.0,0.0,Price Consistent
4,Kayali,Freedom Musk Santal 34,USA,105.0,105.0,0.0,Price Consistent
5,Jo Malone,English Pear & Freesia,USA,170.0,170.0,0.0,Price Consistent
6,Jo Malone,Myrrh & Tonka,USA,160.0,160.0,0.0,Price Consistent
7,Maison Margiela Replica,By the Fireplace,USA,170.0,170.0,0.0,Price Consistent
8,Maison Margiela Replica,Jazz Club,USA,170.0,170.0,0.0,Price Consistent
9,Byredo,Mojave Ghost,USA,235.0,192.0,43.0,Own site is Higher


Flaging negative gaps — retailer selling cheaper than own site:

In [95]:
brand_risk=gap_summary[gap_summary['Gap']>0]

In [96]:
print("Brand positioning risk — own site priced higher than retailer:")
brand_risk[["Brand", "Fragrance_Name", "Market", "Own_Site_Price", "Retailer_Price", "Gap", "Gap_Direction"]]

Brand positioning risk — own site priced higher than retailer:


,Brand,Fragrance_Name,Market,Own_Site_Price,Retailer_Price,Gap,Gap_Direction
2,Kayali,Oudgasm Tobacco Oud 04,USA,140.0,119.0,21.0,Own site is Higher
9,Byredo,Mojave Ghost,USA,235.0,192.0,43.0,Own site is Higher
10,Byredo,Gypsy Water,USA,235.0,157.0,78.0,Own site is Higher
11,Lattafa,Khamrah,USA,49.0,27.0,22.0,Own site is Higher
12,Lattafa,Yara,USA,44.0,22.0,22.0,Own site is Higher
23,Maison Margiela Replica,Jazz Club,UK,127.0,100.0,27.0,Own site is Higher
24,Byredo,Mojave Ghost,UK,155.0,150.0,5.0,Own site is Higher
25,Byredo,Gypsy Water,UK,155.0,150.0,5.0,Own site is Higher
39,Lattafa,Khamrah,UAE,169.0,140.0,29.0,Own site is Higher
40,Lattafa,Yara,UAE,119.0,116.0,3.0,Own site is Higher


In [97]:
gap_summary["Brand_Fragrance"] = gap_summary["Brand"] + "\n" + gap_summary["Fragrance_Name"]

In [99]:
gap_summary["Brand_Fragrance"] = gap_summary["Brand"] + " " + gap_summary["Fragrance_Name"]

In [102]:
# Combine all 3 market dataframes but only keep USD best price
luxury_df = usa_df[["Brand", "Fragrance_Name", "Type", "Best_price_per_ml_USD"]].copy()
luxury_df = luxury_df[luxury_df["Best_price_per_ml_USD"].notna()]

In [103]:
luxury_df.to_excel('Luxury_positioning.xlsx')

<ipython-input-103-af36d972ee27>:1: UserWarning: Pandas requires version '1.4.3' or newer of 'xlsxwriter' (version '1.2.9' currently installed).
  luxury_df.to_excel('Luxury_positioning.xlsx')


In [106]:
usa_df.to_excel('usa_pricing_cleaned.xlsx')
uk_df.to_excel('uk_pricing_cleaned.xlsx')
uae_df.to_excel('uae_pricing_cleaned.xlsx')
gap_summary.to_excel('pricing_gap_summary.xlsx')

<ipython-input-106-407ee692e4b4>:1: UserWarning: Pandas requires version '1.4.3' or newer of 'xlsxwriter' (version '1.2.9' currently installed).
  usa_df.to_excel('usa_pricing_cleaned.xlsx')
<ipython-input-106-407ee692e4b4>:2: UserWarning: Pandas requires version '1.4.3' or newer of 'xlsxwriter' (version '1.2.9' currently installed).
  uk_df.to_excel('uk_pricing_cleaned.xlsx')
<ipython-input-106-407ee692e4b4>:3: UserWarning: Pandas requires version '1.4.3' or newer of 'xlsxwriter' (version '1.2.9' currently installed).
  uae_df.to_excel('uae_pricing_cleaned.xlsx')
<ipython-input-106-407ee692e4b4>:4: UserWarning: Pandas requires version '1.4.3' or newer of 'xlsxwriter' (version '1.2.9' currently installed).
  gap_summary.to_excel('pricing_gap_summary.xlsx')


In [107]:
pricing_df.to_excel('Pricing_Data_Clean.xlsx')

<ipython-input-107-636bcb61b233>:1: UserWarning: Pandas requires version '1.4.3' or newer of 'xlsxwriter' (version '1.2.9' currently installed).
  pricing_df.to_excel('Pricing_Data_Clean.xlsx')
